torch/_dynamo/variables/builder.py

```diff
@@ -853,6 +856,43 @@ def wrap_tensor(self, value: torch.Tensor):
 
         return tensor_variable
 
+    def wrap_numpy_ndarray(self, value):
+        assert isinstance(value, np.ndarray)
+        arg_proxy = self.tx.output.root_tracer.create_graph_input(
+            re.sub(r"[^a-zA-Z0-9]+", "_", self.name), type(value)
+        )
+        source = self.get_source()
+        tensor_value = torch.from_numpy(value)
+
+        from ..utils import to_torch_np_ndarray
+
+        numpy_ndarray_variable = wrap_fx_proxy_cls(
+            target_cls=NumpyNdarrayVariable,
+            tx=self.tx,
+            proxy=self.tx.output.create_proxy(
+                "call_function",
+                to_torch_np_ndarray,
```

Comment:
This is technically a problem with the previous PR, but I didn't notice it because it is so much easier to do it this way. One thing that is a bit unexpected about this code here is that it means that in the Dynamo produced FX graph, we will have torch_np.ndarray flowing through as intermediates rather than just Tensor. One consequence of this is that it is harder for downstream tools to work directly with the resulting IR.

However, I am a bit unsure how much of a problem this actually is in practice. We then normalize the IR with make_fx, and under the assumption that all of torch_np is traceable (which it should be), the torch_np.ndarray can be made to evaporate again. So maybe this is fine for now. It seems worth some discussion in the code though.

Reply:
torch_np.ndarray is just a simple layer wrapping tensor so hopefully this is fine. Any guidance on how to verify this approach is fine after make_fx?